components/server/src/workspace/workspace-factory.ts

```diff
@@ -188,12 +189,13 @@ export class WorkspaceFactory {
     }
 
     protected generateWorkspaceId(): string {
-        var uuid
-        do {
-            uuid = uuidv4()
-        }
-        while (uuid.charAt(0).match("[0-9]") != null)   // No numbers as first char, as we use this id as DNS name
-        return uuid
+        const randomName: string = uniqueNamesGenerator({
+            dictionaries: [adjectives, colors, animals],
```

Comment:
**thought**: I think using only colors and animals in conjunction with random hash in the end could suffice. This can make items more memorable for the user. 💭 

| CURRENT | PROPOSED |
|-|-|
| ambitious-blue-hamster-155799b3 | blue-hamster-155799b3 |

**question**: Even in the rare case of having the same pair being generated the hash can still be different, right?

Plus, given the ephemeral nature of environments that should not be a problem. In the end, we just need to introduce enough entropy to avoid as much as possible having the same color-animal pair being used.

---

**idea**: We could use later on these generated names for creating shorter URLs if we want, which could be useful or introduce friendlier URLs to share environments, snapshots, and more.

| Environment URL (Example) (BEFORE) | Environment URL (Example) (AFTER) |
|-|-|
| `b9b5e28a-9198-4f4d-8e85-07c864d88026.ws-eu03.gitpod.io` | `blue-hamster-155799b3.gitpod.io` |

Reply:
> **thought**: I think using only colors and animals in conjunction with random hash in the end could suffice. This can make items more memorable for the user. 💭

Yes, let's do that.

> 
> Plus, given the ephemeral nature of environments that should not be a problem. In the end, we just need to introduce enough entropy to avoid as much as possible having the same color-animal pair being used.

Environments are not technically ephemeral and we don't want to test whether a generated ID is unique, because this would be expensive. But I think we can add sufficient characters to the suffix. So the first two names are for humans and the rest is to make them technically unique.